Documentation/ExtensionArchitecture/BestPractises/ExtensionLoadingOrder.rst

```diff
@@ -0,0 +1,75 @@
+..  include:: /Includes.rst.txt
+..  index:: Extension development; Software Design Principles
+..  _extension-loading-order:
+
+=======================
+Extension loading order
+=======================
+
+In TYPO3, the order in which extensions are loaded can impact system behavior.
+This is especially important when an extension overrides, extends, or modifies
+the functionality of another. TYPO3 initializes extensions in a defined order,
+and if dependencies are not loaded beforehand, it can lead to unintended
+behavior.
+
+..  _extension-loading-order-composer:
+
+Composer-based installations
+----------------------------
+
+In Composer-based installations, extensions and dependencies are
+installed based on dependencies specified in the
+:file:`composer.json <extension-composer-json>` file.
+
+For example, if an extension relies on or modifies functionality provided by
+the :php:`ext:felogin` system extension, the dependency should be defined
+as follows:
+
+..  literalinclude:: _snippets/_require-composer.json
+    :language: json
+    :caption: Excerpt of EXT:my_extension/composer.json
+
+This dependency will ensure that TYPO3 loads the extension **after** the
+:php:`ext:felogin` system extension.
+
+Extensions can also "suggest" instead of "require" other extensions. The 3rd
+party extension will then only optionally add further functionality.
+
+In this case, suggested extensions will be loaded **before** the current
+extensions, just as if it was required. So this acts as a kind of "loading
+order suggestion".
```

Comment:
word play. composer (and TYPO3 ext_emconf.php) does not know anything about "optional" dependencies, and TYPO3 already used "suggest" as optional dependency ordering before composer. Based on the absense of "optional" dependencies, this has been adopted.

> "loading order suggestion"

suggerates that the "ordering" is only a suggestion which can be discarded, which it is not. Maybe use "loading order for optional dependency/extension".

Reply:
I reformulated the section